crypto/params.c

```diff
@@ -1009,57 +1009,61 @@ int OSSL_PARAM_set_double(OSSL_PARAM *p, double val)
             *(double *)p->data = val;
             return 1;
         }
-    } else if (p->data_type == OSSL_PARAM_UNSIGNED_INTEGER
-               && val == (ossl_uintmax_t)val) {
```

Comment:
Removing this loses the test for the value being integral.

I.e. val = 3.2 will convert to integer (3 or 4 depending on rounding mode) without error.